tests/registries/test_requires_finalizer.py

```diff
@@ -0,0 +1,152 @@
+import pytest
+
+import kopf
+from kopf.reactor.registries import Resource, GlobalRegistry
+
+
+OBJECT_BODY = {
+    'apiVersion': 'group/version',
+    'kind': 'singular',
+    'metadata': {
+        'name': 'test',
+        'labels': {
+            'key': 'value',
+        },
+        'annotations': {
+            'key': 'value',
+        }
+    }
+}
+
+
+@pytest.mark.parametrize('optional', [
+    pytest.param(True, id='optional'),
+    pytest.param(False, id='mandatory'),
+])
+def test_requires_finalizer_deletion_handler(optional):
+    registry = GlobalRegistry()
+    resource = Resource('group', 'version', 'plural')
+
+    @kopf.on.delete('group', 'version', 'plural',
+                    registry=registry, optional=optional)
+    def fn(**_):
+        pass
+
+    assert registry.requires_finalizer(resource=resource, body=OBJECT_BODY) is not optional
```

Comment:
This construct is a bit counter-intuitive. Maybe, parametrize with 2 values: `@pytest.mark.parametrize('optional, expected'...`, and then compare with `== expected`.

Also, it is better to put the actual result of a function call to a variable, and then use it in the assertion. Pytest then renders the failed assertions nice (and not so nice in some case when the function call is used: because double-calling for rendering can have side-effects).